plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -1478,6 +1478,7 @@ public synchronized void progressChanged(ProgressEvent progressEvent)
         private int bufferSize;
 
         private boolean failed;
+        private boolean multipartUploadStarted;
```

Comment:
there are effectively 2 threads here

- calling thread (one thread; as it's not meant to be thread-safe)
- the background uploading thread (because we never submit another task to executor until previous finished)

let's document, for each field, which thread(s) use it

Reply:
done.

And dropped other commits, replacing with commit message